HotelBooking/HotelBooking.BusinessLogic/Services/Implementation/ReservationService.cs

```diff
@@ -37,4 +47,49 @@ public async Task<ReservationDto> GetReservation(int id)
 
         return dto;
     }
+
+    public async Task AddReservation(CreateReservationDto createReservationDto)
+    {
+        await _validationService.Validate(createReservationDto);
+
+        var roomInfos = createReservationDto.Rooms;
+
+        var rooms = new List<Room>();
+        decimal price = 0;
+
+        foreach (var roomInfo in roomInfos)
+        {
+            var room = await _roomRepository.GetRoom((DataAccessLayer.Entities.RoomType)roomInfo.Type, roomInfo.Capacity, createReservationDto.DateFrom, createReservationDto.DateTo);
+            
+            if(room is null)
+            {
+                throw new NotFoundException("Selected rooms are not available on the specified date.");
```

Comment:
Tutaj raczej nie powinien lecieć NotFoundException. NotFoundException mapuje nam się na statusCode 404. Teraz zobacz, ktoś wywołuje POST /reservations i dostaje 404. 404 oznacza brak zasobu, tutaj okej nie mamy pokoi, ale błąd jest mylący. On ma uzasadnienie gdy chcemy pobrać rezerwację, jej nie ma i klient dostaje 404 not found. A tutaj klient dodaje rezerwację, której wiadomo, że nie ma i dostaje 404. Te pokoje na które chce rezerwować też są, ten zasób jest, tylko jest on niedostępny akurat w tym przedziale dat, ale to już jest kwesia logiki biznesowej, a nie faktycznego braku zasobu. Myślę, że już lepiej pasowałby status 400. Tym bardziej jest to niekonsekwente, ponieważ kilka linii kodu niżej gdy nie ma gościa rzucasz BadRequestException co mapuje się nam na 400.  Podsumowując lepiej byłoby tutaj rzucić BadRequestException